dora/core/server/worker/src/main/java/alluxio/worker/dora/PagedDoraWorker.java

```diff
@@ -590,32 +592,40 @@ public ListenableFuture<List<RouteFailure>> copy(List<Route> routes, UfsReadOpti
       WriteOptions writeOptions) {
     List<ListenableFuture<Void>> futures = new ArrayList<>();
     List<RouteFailure> errors = Collections.synchronizedList(new ArrayList<>());
-    FileSystemContext fsContext = FileSystemContext.create(Configuration.global());
+
     for (Route route : routes) {
-      FileSystem srcFs = FileSystem.Factory.create(fsContext,
-          FileSystemOptions.create(Configuration.global(),
-              Optional.of(new UfsFileSystemOptions(new AlluxioURI(route.getSrc()).getRootPath()))));
-      FileSystem dstFs = FileSystem.Factory.create(fsContext,
-          FileSystemOptions.create(Configuration.global(),
-              Optional.of(new UfsFileSystemOptions(new AlluxioURI(route.getDst()).getRootPath()))));
-
-      ListenableFuture<Void> future = Futures.submit(() -> {
-        try {
-          if (readOptions.hasUser()) {
-            AuthenticatedClientUser.set(readOptions.getUser());
-          }
-          CopyHandler.copy(route, writeOptions, srcFs, dstFs);
-        } catch (Exception t) {
-          AlluxioRuntimeException e = AlluxioRuntimeException.from(t);
-          RouteFailure.Builder builder =
-              RouteFailure.newBuilder().setRoute(route).setCode(e.getStatus().getCode().value());
-          if (e.getMessage() != null) {
-            builder.setMessage(e.getMessage());
+      UnderFileSystem srcUfs = mUfsManager.getOrAdd(new AlluxioURI(route.getSrc()),
```

Comment:
is this going to keep filling up the unfsfsmap in ufsmanger?

Reply:
yes, this not gonna be a long term solution